src/modules/CC1101.cpp

```diff
@@ -480,11 +480,12 @@ uint8_t CC1101::getLQI() {
 
 size_t CC1101::getPacketLength(bool update) {
   if(!_packetLengthQueried && update) {
-
-    uint8_t format = SPIgetRegValue(CC1101_REG_PKTCTRL0, 1, 0);
-    if (format == CC1101_LENGTH_CONFIG_VARIABLE) {
+    if (_packetLengthConfig == CC1101_LENGTH_CONFIG_VARIABLE) {
+      _packetLength = _mod->SPIreadRegister(CC1101_REG_FIFO);
+    } else {
       _packetLength = _mod->SPIreadRegister(CC1101_REG_FIFO);
```

Comment:
I think we should read the length from CC1101_REG_PKTLEN for fixed packets, right?

Reply:
Yeah, sorry. The upcoming one should do. I've started integrating the same changes in other modules, too. Will push in the following hours. For some of them I have hardware that I can test.